pkg/daemon/controller/controller.go

```diff
@@ -104,6 +106,7 @@ type ServiceController struct {
 	nodeName       string
 	watchNamespace string
 	psCmdRunner    powershell.CommandRunner
+	mu             sync.Mutex
```

Comment:
Why do we need this lock? Ideally whenever introducing a lock, the reason for doing so should be mentioned in the commit and in the code.